pkg/testing/tools/artifacts_api_test.go

```diff
@@ -0,0 +1,222 @@
+				  "url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-windows-x86_64.zip",
+				  "sha_url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-windows-x86_64.zip.sha512",
+				  "asc_url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-windows-x86_64.zip.asc",
+				  "type": "zip",
+				  "architecture": "x86_64",
+				  "os": [
+					"windows"
+				  ]
+				},
+				"elastic-agent-8.9.0-SNAPSHOT-aarch64.rpm": {
+				  "url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-aarch64.rpm",
+				  "sha_url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-aarch64.rpm.sha512",
+				  "asc_url": "https://snapshots.elastic.co/8.9.0-1ee08db3/downloads/beats/elastic-agent/elastic-agent-8.9.0-SNAPSHOT-aarch64.rpm.asc",
+				  "type": "rpm",
+				  "architecture": "aarch64",
+				  "attributes": {
+					"include_in_repo": "true",
+					"oss": "false"
+				  }
+				}
+			  },
+			  "dependencies": []
+			}
+		  },
+		  "start_time": "Thu, 1 Jun 2023 00:17:37 GMT",
+		  "release_branch": "master",
+		  "prefix": "",
+		  "end_time": "Thu, 1 Jun 2023 04:20:22 GMT",
+		  "manifest_version": "2.1.0",
+		  "version": "8.9.0-SNAPSHOT",
+		  "branch": "master",
+		  "build_id": "8.9.0-1ee08db3",
+		  "build_duration_seconds": 14565
+		},
+		"manifests": {
+		  "last-update-time": "Thu, 01 Jun 2023 08:36:07 UTC",
+		  "seconds-since-last-update": 208
+		}
+	  }
+	`
+)
+
+func TestDefaultArtifactAPIClientErrorHttpStatus(t *testing.T) {
+
+	httpErrorCodes := []int{
+		http.StatusBadRequest,
+		http.StatusNotFound,
+		http.StatusForbidden,
+		http.StatusTeapot,
+		http.StatusInternalServerError,
+	}
+
+	for _, httpErrorCode := range httpErrorCodes {
+		t.Run(fmt.Sprintf("HTTPCode%d", httpErrorCode), func(t *testing.T) {
+
+			errorHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(httpErrorCode)
+			})
+			testSrv := httptest.NewServer(errorHandler)
+			defer testSrv.Close()
+
+			aac := NewArtifactAPIClient(WithUrl(testSrv.URL))
+			_, err := aac.GetVersions(context.Background())
+			assert.ErrorIs(t, err, ErrBadHTTPStatusCode, "Expected ErrBadHTTPStatusCode for status code %d", httpErrorCode)
+			_, err = aac.GetBuildsForVersion(context.Background(), "1.2.3-SNAPSHOT")
+			assert.ErrorIs(t, err, ErrBadHTTPStatusCode, "Expected ErrBadHTTPStatusCode for status code %d", httpErrorCode)
+			_, err = aac.GetBuildDetails(context.Background(), "1.2.3", "abcdefg")
+			assert.ErrorIs(t, err, ErrBadHTTPStatusCode, "Expected ErrBadHTTPStatusCode for status code %d", httpErrorCode)
+		})
+	}
+}
+
+func TestDefaultArtifactAPIClient(t *testing.T) {
+
+	cannedRespHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
[Question]
What does `canned` means here?
pretty much any name would do here, even `h`, so I'm wondering what is the meaning of `canned`

Reply:
Just a prepared (canned) response returned by the test http server, as you say the name is not really important... can be changed to whatever

Reply:
ah, ok, I guess I'm just not used to this naming convention